fix pbme on Windows in episode-2

annefou opened the pull request.

Commit:
fix pbme on Windows in episode-2

annefou commented:
- installed combo in `basic-scipy-env` where numba is already installed (otherwise it fails on Windows and python 3.9).
- moved creation of folder from setup to this section 
- install `tensorflow` 2.0 and not 2.1 
- make sure learners/instructors know that on `\` is required for PowerShell only (not on git bash).

Commit:
add instructions to install Miniconda3 on Windows

naoe-tatara committed:
Deleted "from Anaconda" in l.51 and small fixes.

- line 51: I feel "... install the Python 3 version of Miniconda from Anaconda for your OS" is confusing for novices. The linked page is not Anaconda.com, and Anaconda does not have distribution of miniconda, either.
- line 27: deleted "you" before "then this command should return ..." (I believe that it was unintentionally left while editing by the first author).
- line 44: deleted one ")" which was unnecessary.

naoe-tatara committed:
Moved Binder solution at the bottom

Not sure about the heading, but tried to make it as neutral as possible. Feel free to suggest a better heading.

naoe-tatara commented:
I am going to work on adding a sub-section of installation on Mac OS now :)

davidrpugh commented on setup.md:
```diff
@@ -211,31 +211,3 @@ If you want to reverse or “undo” the changes made by `conda init`, then you
 `conda init` command and pass the `--reverse` option. Again, in order for the reversal to take 
 effect you will likely need to close and restart your shell session.
 
-## Workspace Setup
```
Why remove this from the `setup.md` file? Typically SWC and DC lessons will include the creation of any working directory for the lesson at part of the `setup.md` file.

annefou replied:
Mostly because  I found this section a bit hidden; for instance if using binder, we would probably skip this step. And we do not really need a project folder before this episode 2 (episode 1 is mostly about the context/introduction).

But I agree we could move it either at the bottom of episode 1 or at the top of episode 2. What do you think?

annefou replied:
I moved it on top of episode 2. Alternatively, we could move it back to setup but add a table of content (so that everyone understands what steps are required.

davidrpugh replied:
It is important that the section be included in the `setup.md` file. We can also add it in the lessons at the top of episode 2 if you feel it needs the extra visibility.

davidrpugh commented on _episodes/02-working-with-environments.md:
```diff
@@ -350,6 +352,34 @@ Running `ls` (linux) / `dir` (Windows) on your anaconda `envs/` directory will l
 > previously installed Conda environments.
 {: .callout}
 
+## Workspace for Conda environments
```
Why include this here as opposed to the `setup.md` file? Am not opposed to including this in the boy of the lessons but if we do so I think it should be near the beginning of episode 1.

annefou replied:
See answer above for setup.

davidrpugh commented on _episodes/02-working-with-environments.md:
```diff
@@ -313,15 +315,15 @@ $ conda install scikit-learn=0.22
 > in real-world tasks and data science competitions like [Kaggle](https://www.kaggle.com/).
 > 
-> Activate the `machine-learning-env` you created in a previous challenge and use `pip` to install 
+> Activate the `basic-scipy-env` you created in a previous challenge and use `pip` to install 
```
Why do you change from `machine-learning-env` to `basic-scipy-env` for this exercise? If `combo` doesn't install into the `machine-learning-env` on Windows I would prefer to open an issue and figure out why. Combo only makes sense to use with all of the deps in the `machine-learning-env` so when you install into `basic-scipy-env` these deps must be coming from `pip` when `combo` is installed. End result should be the same.

annefou replied:
I think we can add `numba` when creating `machine-learning-env` and it would solve our problem.

`conda create --name machine-learning-env  ipython  matplotlib  pandas  pip  python  scikit-learn numba`

davidrpugh replied:
I would prefer to add `numba` to the `machine-learning-env` and then `pip install combo` into the `machine-learning-env`. Still not sure why this did't work on Windows in the first place though.

annefou replied:
This is already what I did.

davidrpugh replied:
Right! Sorry, I see that now.

Commit:
address review's comment
- added numba in machine-learning-env so that we can pip install numba

naoe-tatara committed:
Added "Mac OSX installation"

I used the same formulation with Windows one for the first line (line 84).
For the second line (line 85), I referred information on https://docs.anaconda.com/anaconda/install/mac-os/ (I tested that this works with Mac OS Catalina).
Regarding the last line (line 86), I tried making something corresponding to the last line for Windows installation (line 81), but if this is deemed unnecessary, please feel free to delete it.

Commit:
move creation folder on top

Commit:
Merge branch 'fix-02' of github.com:annefou/introduction-to-conda-for-data-scientists into fix-02

naoe-tatara commented:
@davidrpugh , we have finished on our side, but do you have anything that we should work on? Thanks for your help!

davidrpugh commented:
@naoe-tatara and @annefou I am happy to merge this PR. I would like to add the Workspace Setup section in the `setup.md` file (in addition to it being included in episode 2). However I can do this myself later.

annefou commented:
@davidrpugh I think you merged before my update on the setup...
